lib/arclight/repository.rb

```diff
@@ -49,15 +49,19 @@ def request_config_present_for_type?(type)
         config&.fetch('request_mappings').present?
     end
 
+    def request_config_for_type(type)
+      request_config_present_for_type?(type) && request_types[type]
```

Comment:
I see below we're also guarding against `type` being `nil`; do we need to do that here at all?

I wonder if it's worth a little refactoring to go ahead and call this method in `request_url_for_type` and `request_mappings_for_type`.